### PR TITLE
feat: added kochava device mode support

### PR DESCRIPTION
**Pull request by desusai7**
## feat: added kochava device mode support

> Added Kochava Device Mode Support

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Development

- [x] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [x]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [x] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: pubspec.yaml [hunk 1]
```diff
@@ -24,8 +24,13 @@ dependencies:
   rudder_integration_appsflyer_flutter: 1.1.9
   rudder_integration_adjust_flutter: 1.1.4
   rudder_integration_leanplum_flutter: 1.1.4
+  rudder_integration_kochava_flutter: 1.0.0
   flutter_plugin_tools: ^0.12.1
 
+dependency_overrides:
```
Comment (bardisg) [inline pubspec.yaml:30]: why is this needed?

**Review by bardisg — APPROVED** [review]

File: pubspec.yaml (hunk 1 above)
Comment (desusai7) [reply pubspec.yaml]: added this while testing and it is not needed, removed it
